applications/asset_tracker_v2/src/cloud/cloud_codec/cloud_codec.h

```diff
@@ -255,29 +259,94 @@ enum cloud_codec_event_type {
 };
 
 struct cloud_codec_evt {
+	/** Cloud codec event type */
 	enum cloud_codec_event_type type;
+	/** New config data if type is CLOUD_CODEC_EVT_CONFIG_UPDATE */
 	struct cloud_data_cfg config_update;
 };
 
+/** @brief Event handler prototype.
+ *
+ * @param[in] evt event type
+ */
 typedef void (*cloud_codec_evt_handler_t)(const struct cloud_codec_evt *evt);
 
+/**
+ * @brief Initialize cloud codec.
+ * @details Is called once in the data module to initialize the cloud codec used.
+ * @param[in] cfg initial config data
+ * @param[in] event_handler handler for events coming from the codec
+ *
+ * @note currently only used for config updates in lwm2m
+ *
+ * @return 0 on success or error code according to errno.h.
+ */
 int cloud_codec_init(struct cloud_data_cfg *cfg, cloud_codec_evt_handler_t event_handler);
 
+/**
+ * @brief Encode cloud codec neighbor cells data.
+ * @details Is called as part of the data_encode() function to encode LTE neighbor cells info.
+ * @param[out] output string buffer for encoding result
+ * @param[in] neighbor_cells neighbor cells data
+ * @return 0 on success or error code according to errno.h.
+ */
 int cloud_codec_encode_neighbor_cells(struct cloud_codec_data *output,
 				      struct cloud_data_neighbor_cells *neighbor_cells);
 
+/**
+ * @brief Encode cloud codec A-GPS request.
+ *
+ * @param[out] output string buffer for encoding result
+ * @param[in] agps_request A-GPS request data
+ * @return 0 on success or error code according to errno.h.
+ */
 int cloud_codec_encode_agps_request(struct cloud_codec_data *output,
 				    struct cloud_data_agps_request *agps_request);
 
+/**
+ * @brief Encode cloud codec P-GPS request.
+ *
+ * @param[out] output string buffer for encoding result
+ * @param[in] pgps_request P-GPS request data
+ * @return 0 on success or error code according to errno.h.
+ */
 int cloud_codec_encode_pgps_request(struct cloud_codec_data *output,
 				    struct cloud_data_pgps_request *pgps_request);
 
+/**
+ * @brief Decode received configuration.
+ *
+ * @param[in] input string buffer with encoded config
+ * @param[in] input_len length of input
+ * @param[out] cfg decoded config
+ * @return 0 on success or error code according to errno.h.
+ */
 int cloud_codec_decode_config(char *input, size_t input_len,
 			      struct cloud_data_cfg *cfg);
 
+/**
+ * @brief Encode current configuration.
+ *
+ * @param[out] output string buffer for encoding result
+ * @param[in] cfg current configuration
```

Comment:
Could specify that the input parameter is a pointer, that goes for the other docs added in this patch as well.